app/assets/javascripts/samples/sample_datatable.js

```diff
@@ -91,23 +96,65 @@ table = $("#samples").DataTable({
         }
     },
     stateSaveCallback: function (settings, data) {
-        // Set a cookie with the table state using the organization id
-        localStorage.setItem('datatables_state/' + $("#samples").attr("data-organization-id"), JSON.stringify(data));
+        // Send an Ajax request to the server with the state object
+        var org = $("#samples").attr("data-organization-id")
+        var user = $("#samples").attr("data-user-id")
+
+        // Save correct data
+        if (loadFirstTime == true) {
```

Comment:
You don't need loadFirstTime variable, you can just check this with if (!_isUndefined(myData)).